clash-lib/src/Clash/Rewrite/Util.hs

```diff
@@ -313,11 +322,14 @@ runRewriteSession :: RewriteEnv
                   -> RewriteMonad extra a
                   -> a
 runRewriteSession r s m =
-  traceIf (_dbgLevel r > DebugSilent)
-    ("Clash: Applied " ++ show (s' ^. transformCounter) ++ " transformations")
-    a
+  traceIf (_dbgLevel r >= DebugCount)
+    ("Clash: Transform counters:\n" ++ showCounters (s' ^. transformCounters)) $
+    traceIf (_dbgLevel r > DebugSilent)
+      ("Clash: Applied " ++ show (s' ^. transformCounter) ++ " transformations")
+      a
   where
     (a,s',_) = runR m r s
+    showCounters = unlines . map (\(nm,cnt) -> nm ++ ": " ++ show (Monoid.getSum cnt)) . sortOn snd . Map.toList
```

Comment:
It would be nice if we could print a MarkDown table so we can paste it in a GitHub comment. We'd need something like https://pypi.org/project/tabulate/. I wouldn't mind if this gets postponed to a later PR though.

Reply:
Maybe https://hackage.haskell.org/package/boxes ?

Reply:
Leaving this for a future PR, it's just as easy to read a list of `transformation: count` then it is to read a table IMO. I can't imagine there's a nice way to hack in prettyprinting for just one type of debugging message too...